File: frontend/src/types/portfolio.ts
export interface Portfolio {

    id: number;

    userId: number;

    title: string;

    description: string;

    projectUrl: string;

    imageUrl: string;

}


export interface PortfolioRequest {

    title: string;

    description: string;

    projectUrl: string;

    imageUrl: string;

}

File: backend/src/main/java/com/campusos/backend/portfolio/entity/Portfolio.java
package com.campusos.backend.portfolio.entity;

import com.campusos.backend.user.entity.User;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "portfolios", uniqueConstraints = {
        @UniqueConstraint(columnNames = "user_id")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Portfolio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(length = 2000)
    private String description;

    private String projectUrl;

    private String imageUrl;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

}